app/models/user.rb

```diff
@@ -28,33 +32,56 @@ class User < ApplicationRecord
   devise :database_authenticatable, :registerable,
          :recoverable, :rememberable, :validatable
 
-  has_many :comments, foreign_key: :author_id
+  has_many :comments, foreign_key: :author_id, dependent: :destroy
 
-  has_many :sent_follow_requests, foreign_key: :sender_id, class_name: "FollowRequest"
+  has_many :sent_follow_requests, foreign_key: :sender_id, class_name: "FollowRequest", dependent: :destroy
   
   has_many :accepted_sent_follow_requests, -> { accepted }, foreign_key: :sender_id, class_name: "FollowRequest"
   
-  has_many :received_follow_requests, foreign_key: :recipient_id, class_name: "FollowRequest"
+  has_many :received_follow_requests, foreign_key: :recipient_id, class_name: "FollowRequest", dependent: :destroy
   
   has_many :accepted_received_follow_requests, -> { accepted }, foreign_key: :recipient_id, class_name: "FollowRequest"
 
-  has_many :likes, foreign_key: :fan_id
+  has_many :pending_received_follow_requests, -> { pending }, foreign_key: :recipient_id, class_name: "FollowRequest"
 
-  has_many :own_photos, foreign_key: :owner_id, class_name: "Photo"
+  has_many :likes, foreign_key: :fan_id, dependent: :destroy
+
+  has_many :own_photos, foreign_key: :owner_id, class_name: "Photo", dependent: :destroy
 
   has_many :liked_photos, through: :likes, source: :photo
 
   has_many :leaders, through: :accepted_sent_follow_requests, source: :recipient
   
   has_many :followers, through: :accepted_received_follow_requests, source: :sender
 
+  has_many :pending, through: :pending_received_follow_requests, source: :sender
+
   has_many :feed, through: :leaders, source: :own_photos
 
   has_many :discover, through: :leaders, source: :liked_photos
 
-  validates :username, presence: true, uniqueness: true
+  validates :username,
+    presence: true,
+    uniqueness: true,
+    format: { 
+      with: /\A[\w_\.]+\z/i,
+      message: "can only contain letters, numbers, periods, and underscores"
+    }
+
+  validates :website, url: { allow_blank: true }
 
   scope :past_week, -> { where(created_at: 1.week.ago...) }
 
   scope :by_likes, -> { order(likes_count: :desc) }
+
+  before_validation :ensure_website_has_scheme
+
+  def ensure_website_has_scheme
+    if website.present? &&
+      !website.starts_with?("http://") &&
+      !website.starts_with?("https://")
+
```

Comment:
This is very explicit.

I think this could be condensed with Regex.

 ```suggestion
      !website.starts_with?(/^https?:\/\//)
```

Reply:
I agree that it could be more concise — I initially had it as `!website.starts_with?("http")`, which I _think_ works just as well, but after I spent more than 3 minutes wondering about it I decided that I prefer explicitness.

Reply:
> I'm not sure your version is as intention-revealing as mine to all readers

I agree that your way it is easier to tell what you intend to check in the expression. In regards to checking the format of something, I think Regex is used often enough and this one isn't _too_ ugly that I wanted to mention it.